addons/mail/static/src/models/thread_cache/thread_cache.js

```diff
@@ -156,6 +156,10 @@ function factory(dependencies) {
                 );
             }
             messages = messages.concat(newerMessages);
+            // Pending messages
+            if (this.thread.queue.length > 0) {
+                messages = messages.concat(this.thread.queue);
+            }
```

Comment:
Does it hurt to concat if the array is empty? Feels like an unnecessary condition.

Reply:
It's doesn't.
But if (for some reason) the value is `null` or `undefined`, it will be concatenated  `[ 'test', null ]`.
I don't think it's an issue here any way.